src/silx/io/dictdump.py

```diff
@@ -334,20 +347,28 @@ def _iter_treedict(attributes=False):
                         # Delete the existing dataset
                         if update_mode != "replace":
                             if not is_dataset(h5f[h5name]):
-                                logger.info(f'Cannot overwrite {h5f.file.filename}::{h5f[h5name].name} with update_mode="{update_mode}"')
+                                logger.info(
+                                    f'Cannot overwrite {h5f.file.filename}::{h5f[h5name].name} with update_mode="{update_mode}"'
+                                )
                                 continue
                             attrs_backup = dict(h5f[h5name].attrs)
                         del h5f[h5name]
 
                 # Create dataset
                 # can't apply filters on scalars (datasets with shape == ())
-                if data.shape == () or create_dataset_args is None:
-                    h5f.create_dataset(h5name,
-                                       data=data)
-                else:
-                    h5f.create_dataset(h5name,
-                                       data=data,
-                                       **create_dataset_args)
+                try:
+                    if data.shape == () or create_dataset_args is None:
+                        h5f.create_dataset(h5name, data=data)
+                    else:
+                        h5f.create_dataset(h5name, data=data, **create_dataset_args)
+                except Exception as e:
```

Comment:
No way to have a stricter selection of exceptions?

Reply:
I reduced it to `TypeError` only. Ok?

Reply:
Yes :+1: